qiskit/algorithms/gradients/lin_comb_estimator_gradient.py

```diff
@@ -0,0 +1,126 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Gradient of probabilities with linear combination of unitaries (LCU)
+"""
+
+from __future__ import annotations
+
+from typing import Sequence
+
+import numpy as np
+
+from qiskit.circuit import Parameter, ParameterExpression, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info import Pauli
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_estimator_gradient import BaseEstimatorGradient
+from .estimator_gradient_result import EstimatorGradientResult
+from .utils import make_lin_comb_gradient_circuit
+
+Pauli_Z = Pauli("Z")
+
+
+class LinCombEstimatorGradient(BaseEstimatorGradient):
+    """Compute the gradients of the expectation values.
+    This method employs a linear combination of unitaries [1].
+
+    **Reference:**
+    [1] Schuld et al., Evaluating analytic gradients on quantum hardware, 2018
+    `arXiv:1811.11184 <https://arxiv.org/pdf/1811.11184.pdf>`_
+    """
+
+    def __init__(self, estimator: BaseEstimator, **run_options):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in ``run`` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+        """
+        self._gradient_circuits = {}
+        super().__init__(estimator, **run_options)
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
```

Comment:
The LCU gradients currently don't seem to work with `PauliSumOp`:
```
  ...
  File "/Users/jul/opt/miniconda3/envs/py10/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/jul/Qiskit/qiskit-terra/qiskit/algorithms/gradients/lin_comb_estimator_gradient.py", line 76, in _run
    observable_ = observable.expand(Pauli_Z)
AttributeError: 'PauliSumOp' object has no attribute 'expand'
```

Reply:
It should be converted into another operator class before expand.

Reply:
I don't know about operators... What should I do to convert into another operator class?

Reply:
You can use `init_observable` from the primitive utils to convert a valid input type to a `SparsePauliOp` 🙂